pkg/controller/directimagemigration/validation.go

```diff
@@ -0,0 +1,265 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package directimagemigration
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"reflect"
+
+	liberr "github.com/konveyor/controller/pkg/error"
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	migref "github.com/konveyor/mig-controller/pkg/reference"
+	kapi "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Types
+const (
+	InvalidSourceClusterRef               = "InvalidSourceClusterRef"
+	InvalidDestinationClusterRef          = "InvalidDestinationClusterRef"
+	InvalidSourceCluster                  = "InvalidSourceCluster"
+	InvalidDestinationCluster             = "InvalidDestinationCluster"
+	SourceClusterNotReady                 = "SourceClusterNotReady"
+	DestinationClusterNotReady            = "DestinationClusterNotReady"
+	MissingSourceClusterRegistryPath      = "MissingSourceClusterRegistryPath"
+	MissingDestinationClusterRegistryPath = "MissingDestinationClusterRegistryPath"
+	NsListEmpty                           = "NamespaceListEmpty"
+	NsNotFoundOnSourceCluster             = "NamespaceNotFoundOnSourceCluster"
+
+	Running   = "Running"
+	Failed    = "Failed"
+	Succeeded = "Succeeded"
+)
+
+// Reasons
+const (
+	NotFound    = "NotFound"
+	NotSet      = "NotSet"
+	NotDistinct = "NotDistinct"
+	NotReady    = "NotReady"
+)
+
+// Categories
+const (
+	Critical = migapi.Critical
+	Advisory = migapi.Advisory
+)
+
+// Statuses
+const (
+	True  = migapi.True
+	False = migapi.False
+)
+
+// Validate the image migration resource
+func (r ReconcileDirectImageMigration) validate(imageMigration *migapi.DirectImageMigration) error {
+	err := r.validateSrcCluster(imageMigration)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	err = r.validateDestCluster(imageMigration)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	// Migrated namespaces.
+	err = r.validateNamespaces(imageMigration)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	return nil
+}
+
+func (r ReconcileDirectImageMigration) validateSrcCluster(imageMigration *migapi.DirectImageMigration) error {
+	ref := imageMigration.Spec.SrcMigClusterRef
+
+	// Not Set
+	if !migref.RefSet(ref) {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     InvalidSourceClusterRef,
+			Status:   True,
+			Reason:   NotSet,
+			Category: Critical,
+			Message:  "The directImageMigration.srcMigClusterRef must reference name and namespace for a valid `MigCluster`",
```

Comment:
NIT: I would rather use `spec.srcMigClusterRef` because directImageMigration is already the name of the CR when user does a -oyaml to get these conditions